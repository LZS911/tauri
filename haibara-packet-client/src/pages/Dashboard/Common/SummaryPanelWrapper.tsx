import { genDesc, genTrend, SummaryCommonPanelProps } from '.';
import ThemeBase from '../../../components/ThemeBase';
import { formatPrice } from '../../../utils/math';

const SummaryPanelWrapper: React.FC<SummaryCommonPanelProps> = ({
  totalData = 0,
  trend,
  trendData = 0,
  title,
  addSymbol = false,
}) => {
  return (
    <ThemeBase.Paper className="mt-4 max-w-[100%] basis-[100%] rounded-md border border-stone-200 p-5 dark:border-stone-700 md:max-w-[48%] md:basis-[48%] lg:max-w-[32%] lg:basis-[32%] xl:max-w-[23%] xl:basis-[23%]">
      <div className="opacity-70 hover:cursor-pointer">{title}</div>
      <div className="mb-4 mt-2 flex">
        <div className="mr-4 text-xl font-bold">
          {formatPrice(totalData, addSymbol)}
        </div>
        {genTrend(trend, trendData)}
      </div>
      <div className="overflow-hidden text-ellipsis whitespace-nowrap text-xs opacity-70">
        {genDesc(trend, trendData)}
      </div>
    </ThemeBase.Paper>
  );
};

export default SummaryPanelWrapper;
