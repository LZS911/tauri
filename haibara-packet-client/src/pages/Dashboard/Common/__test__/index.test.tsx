import { render } from '@testing-library/react';
import { DashboardPanelEnum } from '../..';
import SummaryPanelWrapper, {
  DataTrendEnum,
  genDesc,
  genTrend,
} from '../index';

const title = 'test';

describe('test Common', () => {
  test('should match snapshot for genDesc', () => {
    expect(genDesc()).toMatchSnapshot();
    expect(genDesc(DataTrendEnum.fall)).toMatchSnapshot();
    expect(genDesc(undefined, 10)).toMatchSnapshot();
    expect(genDesc(DataTrendEnum.fall, 20)).toMatchSnapshot();
  });

  test('should match snapshot for genTrend', () => {
    expect(genTrend()).toMatchSnapshot();
    expect(genTrend(DataTrendEnum.fall)).toMatchSnapshot();
    expect(genTrend(undefined, 10)).toMatchSnapshot();
    expect(genTrend(DataTrendEnum.fall, 20)).toMatchSnapshot();
  });

  test('should match snapshot for SummaryPanelWrapper', () => {
    const { container } = render(
      <SummaryPanelWrapper
        trend={DataTrendEnum.fall}
        title={title}
        key={DashboardPanelEnum.TotalOrder}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
