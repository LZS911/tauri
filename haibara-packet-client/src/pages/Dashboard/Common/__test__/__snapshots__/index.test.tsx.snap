// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Common should match snapshot for SummaryPanelWrapper 1`] = `
<div>
  <div
    class="bg-white dark:bg-darkMode dark:text-white mt-4 max-w-[100%] basis-[100%] rounded-md border border-stone-200 p-5 dark:border-stone-700 md:max-w-[48%] md:basis-[48%] lg:max-w-[32%] lg:basis-[32%] xl:max-w-[23%] xl:basis-[23%]"
  >
    <div
      class="opacity-70 hover:cursor-pointer"
    >
      test
    </div>
    <div
      class="mt-2 mb-4 flex"
    >
      <div
        class="mr-4 text-xl font-bold"
      >
        0
      </div>
      <span
        class="border-yellow-500 bg-[#fff7e0] text-yellow-500 dark:bg-[#342c1a] flex items-center rounded-md border  px-2 text-xs"
      >
        <span
          aria-label="fall"
          class="anticon anticon-fall mr-1"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="fall"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M925.9 804l-24-199.2c-.8-6.6-8.9-9.4-13.6-4.7L829 659.5 557.7 388.3c-6.3-6.2-16.4-6.2-22.6 0L433.3 490 156.6 213.3a8.03 8.03 0 00-11.3 0l-45 45.2a8.03 8.03 0 000 11.3L422 591.7c6.2 6.3 16.4 6.3 22.6 0L546.4 490l226.1 226-59.3 59.3a8.01 8.01 0 004.7 13.6l199.2 24c5.1.7 9.5-3.7 8.8-8.9z"
            />
          </svg>
        </span>
        0%
      </span>
    </div>
    <div
      class="overflow-hidden text-ellipsis whitespace-nowrap text-xs opacity-70"
    />
  </div>
</div>
`;

exports[`test Common should match snapshot for genDesc 1`] = `null`;

exports[`test Common should match snapshot for genDesc 2`] = `null`;

exports[`test Common should match snapshot for genDesc 3`] = `null`;

exports[`test Common should match snapshot for genDesc 4`] = `"dashboard.dataSummary.panelFailDesc"`;

exports[`test Common should match snapshot for genTrend 1`] = `undefined`;

exports[`test Common should match snapshot for genTrend 2`] = `undefined`;

exports[`test Common should match snapshot for genTrend 3`] = `undefined`;

exports[`test Common should match snapshot for genTrend 4`] = `
<span
  className="border-yellow-500 bg-[#fff7e0] text-yellow-500 dark:bg-[#342c1a] flex items-center rounded-md border  px-2 text-xs"
>
  <ForwardRef(FallOutlined)
    className="mr-1"
  />
  2000%
</span>
`;
